arroyo/processing/processor.py

```diff
@@ -398,58 +375,59 @@ def _run_once(self) -> None:
             self.__metrics_buffer.incr_timing(
                 "arroyo.consumer.processing.time", time.time() - start_poll
             )
-            if self.__message is not None:
```

Comment:
I think you need a pretty substantial rebase regarding everything below this line.
I'll review after that.